openapi/openapi_v2_backend_configuration_test.go

```diff
@@ -124,6 +124,302 @@ func TestGetHost(t *testing.T) {
+			},
+		}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(spec, openAPIDocumentURL)
+		Convey("When getDefaultRegion() method is called", func() {
+			_, err := specV2BackendConfiguration.getDefaultRegion()
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldNotBeNil)
+			})
+			Convey("And the error should match the expected message", func() {
+				So(err.Error(), ShouldEqual, "failed to get default region value: service is not multi-region")
+			})
+		})
+	})
+	Convey("Given a specV2BackendConfiguration that is multi-region", t, func() {
+		spec := &spec.Swagger{
+			SwaggerProps: spec.SwaggerProps{
+				Swagger: "2.0",
+			},
+			VendorExtensible: spec.VendorExtensible{
+				Extensions: spec.Extensions{
+					extTfProviderMultiRegionFQDN: "www.${region}.some-backend.com",
+					extTfProviderRegions:         "",
+				},
+			},
+		}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(spec, openAPIDocumentURL)
+		Convey("When getDefaultRegion() method is called", func() {
+			_, err := specV2BackendConfiguration.getDefaultRegion()
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldNotBeNil)
+			})
+			Convey("And the error should match the expected message", func() {
+				So(err.Error(), ShouldEqual, "failed to get default region value: 'x-terraform-provider-regions' extension missing or empty value provided")
+			})
+		})
+	})
+}
+
+func TestIsMultiRegion(t *testing.T) {
+	Convey("Given a specV2BackendConfiguration that is multi-region (with 'x-terraform-provider-regions' extension values being comma separated with spaces)", t, func() {
+		spec := &spec.Swagger{
+			SwaggerProps: spec.SwaggerProps{
+				Swagger: "2.0",
+			},
+			VendorExtensible: spec.VendorExtensible{
+				Extensions: spec.Extensions{
+					extTfProviderMultiRegionFQDN: "www.${region}.some-backend.com",
+					extTfProviderRegions:         "rst1, dub1",
+				},
+			},
+		}
+		openAPIDocumentURL := "www.domain.com"
+		specV2BackendConfiguration, _ := newOpenAPIBackendConfigurationV2(spec, openAPIDocumentURL)
+		Convey("When isMultiRegion() method is called", func() {
+			isMultiRegion, host, regions, err := specV2BackendConfiguration.isMultiRegion()
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+			Convey("Then it should be multi region", func() {
+				So(isMultiRegion, ShouldBeTrue)
+			})
+			Convey("Then host should be the parametrised host", func() {
+				So(host, ShouldEqual, "www.${region}.some-backend.com")
+			})
+			Convey("Then regions should contain the right regions", func() {
+				So(regions, ShouldContain, "rst1")
+				So(regions, ShouldContain, "dub1")
+			})
+		})
+	})
+	Convey("Given a specV2BackendConfiguration that is multi-region (with 'x-terraform-provider-regions' extension values being comma separated with NO spaces)", t, func() {
```

Comment:
This is a lot of code to test the whitespace regex

Reply:
agreed, will be tackling unit test clean up soon - different PR